### PR TITLE
Filter edge lock when getting edge

### PR DESCRIPTION
**Pull request by panda-sheep**
Filter edge lock when getting edge.
Because edge lock in toss and edge are the same except for the last byte.

### CODE REVIEW

File: src/storage/admin/RebuildTagIndexTask.cpp [hunk 1]
```diff
@@ -72,10 +72,6 @@ kvstore::ResultCode RebuildTagIndexTask::buildIndexGlobal(GraphSpaceID space,
 
         auto key = iter->key();
         auto val = iter->val();
-        if (!NebulaKeyUtils::isVertex(vidSize, key)) {
```
Comment (critical27) [inline src/storage/admin/RebuildTagIndexTask.cpp:75]: Remove from `RebuildEdgeIndexTask.cpp` as well

File: src/storage/admin/StatisTask.cpp [hunk 2]
```diff
@@ -185,6 +185,13 @@ StatisTask::genSubTask(GraphSpaceID spaceId,
...
     while (edgeIter && edgeIter->valid()) {
         auto key = edgeIter->key();
+
+        // Because edge lock in toss and edge are the same except for the last byte
+        if (!NebulaKeyUtils::isEdge(vIdLen, key)) {
+            edgeIter->next();
+            continue;
+        }
+
         auto edgeType = NebulaKeyUtils::getEdgeType(vIdLen, key);
```
Comment (critical27) [inline src/storage/admin/StatisTask.cpp:195]: You could remove the useless multi version check in count *

File: src/storage/query/ScanVertexProcessor.cpp [hunk 3]
```diff
@@ -63,9 +63,6 @@ void ScanVertexProcessor::doProcess(const cpp2::ScanVertexRequest& req) {
     RowReaderWrapper reader;
     for (int64_t rowCount = 0; iter->valid() && rowCount < rowLimit; iter->next()) {
         auto key = iter->key();
-        if (!NebulaKeyUtils::isVertex(spaceVidLen_, key)) {
```
Comment (critical27) [inline src/storage/query/ScanVertexProcessor.cpp:66]: Remove it in ScanEdgeProcessor as well

File: src/storage/mutate/AddEdgesProcessor.cpp [hunk 4]
```diff
@@ -392,8 +392,12 @@ AddEdgesProcessor::findOldValue(PartitionID partId, const folly::StringPiece& ra
                    << ", spaceId " << spaceId_;
         return folly::none;
     }
-    if (iter && iter->valid()) {
-        return iter->val().str();
+
+    while (iter && iter->valid()) {
```
Comment (critical27) [inline src/storage/mutate/AddEdgesProcessor.cpp:396]: No need to modify here. #338 will turn the direction. As for out edge, there won't be any lock.

**Comment by critical27** [comment]
Good job, check other place if they call useless isVertex or isEdge

File: src/storage/admin/RebuildTagIndexTask.cpp (hunk 1 above)
Comment (panda-sheep) [reply src/storage/admin/RebuildTagIndexTask.cpp:75]: Good catch!

File: src/storage/mutate/AddEdgesProcessor.cpp (hunk 4 above)
Comment (panda-sheep) [reply src/storage/mutate/AddEdgesProcessor.cpp]: Yeah, 
btw Before merging #338 , please temporarily stop the use of toss

File: src/storage/query/ScanVertexProcessor.cpp (hunk 3 above)
Comment (panda-sheep) [reply src/storage/query/ScanVertexProcessor.cpp:66]: ScanEdgeProcessor  can handle in-edge， so `isEdge  `needs.

File: src/storage/admin/StatisTask.cpp (hunk 2 above)
Comment (panda-sheep) [reply src/storage/admin/StatisTask.cpp:175]: Good point！

File: src/storage/admin/StatisTask.cpp [hunk 5]
```diff
@@ -131,17 +131,15 @@ StatisTask::genSubTask(GraphSpaceID spaceId,
...
-    // Vid  tagId  version
-    // 1     1     2
-    // 1     1     1
-    // 2     2     2   (invalid data)
-    // 2     2     1   (invalid data)
-    // 2     3     1
-    // 3     1     2
+    // Vid  tagId
+    // 1     1
+    // 2     2  (invalid data)
```
Comment (critical27) [inline src/storage/admin/StatisTask.cpp:139]: Tag 2 is dropped? why invalid, better add some comments.

**Review by critical27 — COMMENTED** [review]
LGTM now

File: src/storage/admin/StatisTask.cpp (hunk 5 above)
Comment (panda-sheep) [reply src/storage/admin/StatisTask.cpp]: Has added comments.
invalid tag data, for example, tag data without tag schema.  create tag, insert tag data, then drop tag, so this tag data is invalid.

**Review by critical27 — APPROVED** [review]
Good job

**Review by liuyu85cn — APPROVED** [review]
LGTM

**Pull request merged**